docs/src/design/stack/io_ops.md

```diff
@@ -157,15 +151,57 @@ $$
 Using the above variable, we define the value representing the memory access request as follows:
 
 $$
-u_{mem} = \alpha_0 + \alpha_1 \cdot op_{mem\_write}  + \alpha_3 \cdot s_0 + \alpha_4 \cdot clk + v
+u_{mem} = \alpha_0 + \alpha_1 \cdot op_{mem\_write} + \alpha_2 \cdot ctx + \alpha_3 \cdot s_0 + \alpha_4 \cdot clk + v
 $$
 
 In the above:
 - $op_{mem\_write} $ is the unique [operation label](../chiplets/main.md#operation-labels) of the memory write operation.
 - $s_0$ is the memory address into which the value from the stack is to be saved.
 - $clk$ is the current clock cycle of the VM.
 
-Note that $\alpha_2$ term is skipped because currently memory context value is always $0$.
-
 The effect of this operation on the rest of the stack is:
 * **Left shift** starting from position $1$.
+
+### MSTREAM
+
+The `MSTREAM` operation loads two words from memory, and replaces the top 8 elements of the stack with them, element-wise, in stack order. The memory address from which the words are loaded is stored in the 13th stack element (position 12). The diagram below illustrates this graphically.
+
+![mstream](../../assets/design/stack/io_ops/MSTREAM.png)
+
+After the operation, the memory address is incremented by 2.
+
+$$
+s_{12}' = s_{12} + 2
+$$
+
+To simplify description of the memory access request value, we first define variables for the value that represents the state of memory after the operation:
```

Comment:
minor nit: "define variables for the value" -> "define variables for the values".